netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -86,6 +90,19 @@ public NettyClientHandler(Http2ConnectionEncoder encoder, Http2Connection connec
     connection.remote().maxActiveStreams(0);
     connection.local().allowPushTo(false);
     nextStreamId = connection.local().nextStreamId();
+    connection.addListener(new Http2ConnectionAdapter() {
+      @Override
+      public void onGoAwaySent(int lastStreamId, long errorCode, ByteBuf debugData) {
```

Comment:
This allows us to capture any internal errors that may have occurred in the Netty layer.  Now we have a better way of ensuring that GO_AWAY (both inbound and outbound) result in proper messages.